src/main/java/com/training/assignments/Assignment15.java

```diff
@@ -0,0 +1,23 @@
+package com.training.assignments;
+
+public class Assignment15 {
+    String firstName = "NA";
+    String lastName = "NA";
+    String middleName = "NA";
+
+    public String getFullName() {
+        return "NA";
+    }
+
+    public String getFullName(String firstName) {
+        return firstName;
+    }
+
+    public String getFullName(String firstName, String lastName) {
+        return firstName + " " + lastName;
+    }
+
+    public String getFullName(String firstName, String lastName, String middleName) {
```

Comment:
All your logic should be inside this method, and all the overloaded method should be calling this method finally.